Add `{wrap,peel}_box` functions to `TransparentWrapperAlloc`

WaffleLapkin opened the pull request:
This PR adds the following public API:
```rust
// TransparentWrapperAlloc<Inner: ?Sized>

fn wrap_box(s: Box<Inner>) -> Box<Self> { ... }
fn peel_box(s: Box<Self>) -> Box<Inner> { ... }
```

As a side-note: is it ok that `cargo fmt` changes formatting from the one in the repo?

Lokathor commented:
yeah small format changes aren't a big deal.

Lokathor commented:
Can you throw some simple test cases of this being used in a tests file? Just so that the miri tests can hit the code a little.

WaffleLapkin commented:
I've added tests and also simplified tests that expect values (`{wrap,peel}_vec`), the casts there in the asserts seemed unnecessary, given that we are testing transparent wrapper there, not the casts.

Lokathor commented:
Released in 1.11
